warehouse/manage/views.py

```diff
@@ -288,6 +300,119 @@ def delete_account(self):
         return logout(self.request)
 
 
+@view_defaults(
+    route_name="manage.account.totp-provision",
+    renderer="manage/account/totp-provision.html",
+    uses_session=True,
+    require_csrf=True,
+    require_methods=False,
+    permission="manage:user",
+)
+class ProvisionTOTPViews:
+    def __init__(self, request):
+        self.request = request
+        self.user_service = request.find_service(IUserService, context=None)
+        self.token_service = request.find_service(ITokenService, name="totp_provision")
+
+    @property
+    def default_response(self):
+        return HTTPSeeOther(self.request.route_path("manage.account"))
+
+    @view_config(request_method="GET")
+    def totp_provision(self):
+        two_factor = self.user_service.get_two_factor(self.request.user.id)
+        if two_factor.totp_secret:
+            self.request.session.flash("TOTP already provisioned.", queue="error")
+            return self.default_response
+
+        try:
+            token_data = self.token_service.loads(
+                self.request.cookies.get(TWO_FACTOR_TOTP_PROVISION_COOKIE)
+            )
+        except TokenException:
+            token_data = {
+                "totp_secret": urlsafe_b64encode(otp.generate_totp_secret()).decode()
+            }
+            self.request.response.set_cookie(
+                TWO_FACTOR_TOTP_PROVISION_COOKIE, self.token_service.dumps(token_data)
+            )
+
+        totp_secret = urlsafe_b64decode(token_data["totp_secret"])
+        totp_uri = otp.generate_totp_provisioning_uri(
```

Comment:
We should use the `site.name` configuration variable here for the `issuer_name`.

Reply:
Got it! Will fix tomorrow morning.